prboom2/src/p_inter.c

```diff
@@ -108,6 +108,8 @@ static weapontype_t GetAmmoChange[] = {
     wp_mace
 };
 
+#define autoswitch (allow_incompatibility && !deathmatch && !netgame ? dsda_IntConfig(dsda_config_switch_weapon_on_pickup) : true)
+
```

Comment:
Put () around `allow_incompatibility && !deathmatch && !netgame`. I wouldnt think the ternary captures all those conditions

Reply:
So in my test, this does actually work correctly. (i.e. I tested putting `allow_incompatiblity` at the end) and it still worked as intended.

Lol I will admit it is probably better practice to put `()` around the conditions though, so I'll do it anyway. :P